lib/rouge/lexers/php.rb

```diff
@@ -189,7 +201,8 @@ def self.detect?(text)
       state :string do
         rule %r/"/, Str::Double, :pop!
         rule %r/[^\\{$"]+/, Str::Double
-        rule %r/\\([nrt\"$\\]|[0-7]{1,3}|x[0-9A-Fa-f]{1,2})/,
+        rule /\\u\{[0-9a-fA-F]+\}/, Str::Escape
```

Comment:
Same point here about the `i` modifier.

Reply:
What's best, in fact, among:

* `[0-9a-fA-F]` without the `i` modifier
* `[0-9a-f]` with `i` modifier
* `[[:xdigit:]]` (without `i`)

?

Reply:
My mistake in not realising that the `i` modifier would erroneously match `\u`. I agree the way you had it was better.